homeassistant/components/insteon/config_flow.py

```diff
@@ -129,16 +132,34 @@ async def async_step_plm(self, user_input=None):
         """Set up the PLM modem type."""
         errors = {}
         if user_input is not None:
+            if user_input[CONF_DEVICE] == PLM_MANUAL:
+                return await self.async_step_plm_manually()
             if await _async_connect(**user_input):
                 return self.async_create_entry(title="", data=user_input)
             errors["base"] = "cannot_connect"
         schema_defaults = user_input if user_input is not None else {}
         ports = await async_get_usb_ports(self.hass)
+        if not ports:
+            return await self.async_step_plm_manually()
+        ports[PLM_MANUAL] = "Enter manually"
         data_schema = build_plm_schema(ports, **schema_defaults)
         return self.async_show_form(
             step_id=STEP_PLM, data_schema=data_schema, errors=errors
         )
 
+    async def async_step_plm_manually(self, user_input=None):
+        """Set up the PLM modem type manually."""
+        errors = {}
+        if user_input is not None:
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_plm_manual_schema(**schema_defaults)
```

Comment:
Sorry, what I meant is that you appeared to be checking the condition twice, once in a block, then again as an inline if. i.e. could do this:
```suggestion
        if user_input is not None:
            if await _async_connect(**user_input):
                return self.async_create_entry(title="", data=user_input)
            errors["base"] = "cannot_connect"
            schema_defaults = user_input 
        else:    
            schema_defaults = {}
        data_schema = build_plm_manual_schema(**schema_defaults)
```

Reply:
Similar approach but without the else

Reply:
Even better! :)